app/src/ingest_policy_pdfs.py

```diff
@@ -22,11 +32,93 @@ def _ingest_policy_pdfs(
     embedding_model = app_config.sentence_transformer
     for file in file_list:
         if file.endswith(".pdf"):
+            with smart_open_file(file, "rb") as fin:
+                extract_text_to_fp(fin, output_string, laparams=LAParams(), output_type="text")
+
+                parse_pdf_and_add_to_db(
+                    contents=output_string.getvalue(),
+                    db_session=db_session,
+                    doc_attribs=doc_attribs,
+                    embedding_model=embedding_model,
+                )
+
             logger.info(
                 f"Processing pdf file: {file} at {pdf_file_dir} using {embedding_model}, {db_session}, with {doc_attribs}"
             )
 
 
+def parse_pdf_and_add_to_db(
+    contents: str,
+    db_session: db.Session,
+    doc_attribs: dict[str, str],
+    embedding_model: SentenceTransformer,
+) -> None:
+    # Match header in BEM manual
+    header_pattern = r"(BEM\s\d*\s+\d+\sof\s\d+\s+\w.*)"
+    text_split_by_header = re.split(header_pattern, contents)
+    current_title = ""
+    body_content = ""
+    start_new_section = True
+    for text_contents in text_split_by_header:
+        is_header, contents, start_new_section = get_header_and_is_current_section(
+            text_contents, start_new_section
+        )
+        # Check if we need to start a new section
+        if is_header and start_new_section and body_content != "":
+            current_title = contents
+        else:
+            body_content += f"{contents}\n"
+
+        # If starting a new section and body content is not empty, process and save it
+        if start_new_section and body_content.strip():
+            # Create and add the document to the database
+            document = Document(name=current_title, content=body_content, **doc_attribs)
+            db_session.add(document)
```

Comment:
We want one `Document` per PDF file -- the content of the document should be the full text of the PDF. Then we want to split that text into separate chunks

Reply:
In other words, this code should get pulled out of the loop -- a rough structure could be:

```
pdf_text = extract_text(...)
header = get_header(pdf_text)
document = Document(...)
chunks = get_chunks(pdf_text, document)
```